torchmultimodal/modules/encoders/image_encoder.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Callable, Optional
+
+from torch import nn, Tensor
+from torchmultimodal.modules.layers.transformer import TransformerOutput
+
+
+class VisionTransformer(nn.Module):
+    """
+    General image transformer encoder with embeddings. Similar to ``VisionTransformer`` in torchvision,
+    but more composable. Can be constructed with any user-provided embeddings, encoder, and task head.
+
+    Attributes:
+        embeddings (nn.Module): Module that projects image pixels into embeddings.
+            See :py:class: ImageEmbeddings for interface.
+        encoder (nn.Module): Module for transformer encoder. See :py:class: TransformerEncoder for interface.
+        layernorm (nn.Module, optional): Module for layernorm to be applied after encoder. Defaults to ``None``.
+        pooler (nn.Module, optional): Module for pooler to be applied after layernorm. Defaults to ``None``.
+        weight_init_fn (Callable, optional): function for custom weight initialization of both the transformer
+            encoder and embeddings. See :py:func: init_transformer_weights as an example. Defaults to ``None``.
```

Comment:
The same comment about `py:class` as in https://github.com/facebookresearch/multimodal/pull/249#discussion_r949665113